eslint-plugin/src/rules/no-imports-from-self.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+/**
+ * From: https://github.com/payloadcms/payload/blob/main/packages/eslint-plugin/customRules/no-imports-from-self.js
+ * Copyright (c) Payload CMS, Inc. <info@payloadcms.com>
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
```

Comment:
Let's just keep it to a liner as we've previously done with other references in this repo that add historical value. I like the fact that we're easily able to trace back to the source (which may be updated over time), given that using the package directly is not viable. Thank you!

```suggestion
// From: https://github.com/payloadcms/payload/blob/main/packages/eslint-plugin/customRules/no-imports-from-self.js
```